libs/paymentProcessor.js

```diff
@@ -577,7 +580,7 @@ logger.info(logSystem, logComponent, addressAmounts);
 
 
     var getProperAddress = function(address){
-        if (address.length === 40){
+        if (address.length === 60){
```

Comment:
Whys this changed?

Reply:
That portion of code doesn't even make sense to exist anymore, since miners should be allowed to setup a longer .workername.

It's also broken, and doesn't even change the address to poolconfig pooladdrress.

Cheers

Reply:
I agree with @LEAD-Anoy74 on the issue of the length should not even be checked.

Most pools accept WALLET.WORKER format and the workername can be any length (unless pool limits it)

What I would do is split the address / worker name and then do the length check after, or just rip it out alltogether.

Reply:
Correct

Reply:
The spacing of this is all over the place, but from a quick skim, the function can be kept the same by splitting the address before its passed into getProperAddress. The purpose of this function is as a sanity check to ensure that only a valid base58 bitcoin-type address is passed in and used from that point forward. This change would be breaking that check.